File: apps/web/src/api/pdf.ts
import api from "./client";

// PDF 导入返回的草稿（与后端 PdfImportDraft 一致；不落库，预览后复用 createNote）
export interface PdfImportDraft {
  title: string;
  content: string;
  suggested_tags: string[];
}

export type PdfJobStatus = "pending" | "running" | "done" | "failed";

export interface PdfJobResult {
  job_id: string;
  status: PdfJobStatus;
  progress: number; // 已处理页数
  total: number; // 总页数
  draft: PdfImportDraft | null;
  error: string | null;
}

// 上传 PDF，后台异步转 MD，立即返回 job_id
export async function uploadPdf(file: File): Promise<{ job_id: string }> {
  const form = new FormData();
  form.append("file", file);
  const { data } = await api.post<{ job_id: string }>("/api/pdf/import", form);
  return data;
}

// 轮询任务状态
export async function getPdfJob(jobId: string): Promise<PdfJobResult> {
  const { data } = await api.get<PdfJobResult>(`/api/pdf/jobs/${jobId}`);
  return data;
}

File: apps/api/app/schemas/pdf.py
# PDF 导入：异步任务状态 + 转换草稿模型（与前端对齐）
from typing import Literal

from pydantic import BaseModel


class PdfImportDraft(BaseModel):
    """转换完成的草稿：前端预览编辑后复用 createNote 落库"""
    title: str
    content: str
    suggested_tags: list[str] = []


class PdfJobResult(BaseModel):
    """异步任务状态。done 时 draft 非空；failed 时 error 非空。"""
    job_id: str
    status: Literal["pending", "running", "done", "failed"]
    progress: int = 0      # 已处理页数
    total: int = 0         # 总页数
    draft: PdfImportDraft | None = None
    error: str | None = None


class PdfImportAccepted(BaseModel):
    """POST /api/pdf/import 立即返回：任务已受理"""
    job_id: str
